### PR TITLE
Ikasan 2370 add service to manage indoubt transactions v3

### PR DESCRIPTION
**Pull request by mick-stewart73**

### CODE REVIEW

File: ikasaneip/persistence/src/main/java/org/ikasan/persistence/service/InDoubtTransactionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        return this.inDoubtTransactionDao.getInDoubtTransactions();
+    }
+
+    @Override
+    public InDoubtTransaction getInDoubtTransaction(String transactionName) {
+        return this.inDoubtTransactionDao.getInDoubtTransaction(transactionName);
+    }
+
+    @Override
+    public void commitInDoubtTransaction(String transactionName) {
```
Comment (andrzej-majewski) [inline ikasaneip/persistence/src/main/java/org/ikasan/persistence/service/InDoubtTransactionServiceImpl.java:43]: Should we also add Service to CommitAll and RollBackAll?

File: ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/InDoubtTransactionsApplication.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.stream.Collectors;
+
+
+/**
+ * The InDoubtTransactionsApplication class is a RESTful API controller for handling
+ * in-doubt transactions.
+ */
+@RequestMapping("/rest/transaction/inDoubt")
+@RestController
+public class InDoubtTransactionsApplication
```
Comment (andrzej-majewski) [inline ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/InDoubtTransactionsApplication.java:28]: Should we also add Controller to allow call service to  CommitAll and RollBackAll?

File: ikasaneip/persistence/src/main/java/org/ikasan/persistence/service/InDoubtTransactionServiceImpl.java (hunk 1 above)
Comment (mick-stewart73) [reply ikasaneip/persistence/src/main/java/org/ikasan/persistence/service/InDoubtTransactionServiceImpl.java:43]: @andrzej-majewski I have added feature to commit and roll back all.

File: ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/InDoubtTransactionsApplication.java (hunk 2 above)
Comment (mick-stewart73) [reply ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/InDoubtTransactionsApplication.java:28]: @andrzej-majewski have added feature to commit and rollback all

**Review by andrzej-majewski — APPROVED** [review]

**Pull request merged**